virtcontainers/kata_agent_test.go

```diff
@@ -408,6 +408,62 @@ func TestHandleLocalStorage(t *testing.T) {
 	assert.Equal(t, localMountPoint, expected)
 }
 
+func TestHandleHugepages(t *testing.T) {
+	if os.Getuid() != 0 {
+		t.Skip("Test disabled as requires root user")
+	}
+
+	dir, err := ioutil.TempDir("", "hugepages-test")
+	assert.Nil(t, err)
+	defer os.RemoveAll(dir)
+
+	k := kataAgent{}
+	var mounts []specs.Mount
+	var hugepageLimits []specs.LinuxHugepageLimit
+
+	hugepageDirs := [2]string{"hugepages-1Gi", "hugepages-2Mi"}
+	options := [2]string{"pagesize=1024M", "pagesize=2M"}
+
+	for i := 0; i < 2; i++ {
+		target := path.Join(dir, hugepageDirs[i])
+		err := os.MkdirAll(target, 0777)
+		if err = syscall.Mount("nodev", target, "hugetlbfs", uintptr(0), options[i]); err != nil {
+			fmt.Printf("Unable to mount %s: %v\n", target, err)
+		}
```

Comment:
This looks like it might need a slight rework (since it's test code) to something like this:

```go
err = syscall.Mount("nodev", target, "hugetlbfs", uintptr(0), options[i])
assert.NoError(err, "Unable to mount %s", target")

```